rdt/hyper_transformer.py

```diff
@@ -189,6 +189,36 @@ def update_field_data_types(self, field_data_types):
         self.field_data_types.update(field_data_types)
         self._unfit()
 
+    def update_transformers_by_sdtype(self, sdtype, transformer):
+        """Update the transformers for the specified ``sdtype``.
+
+        Given an ``sdtype`` and a ``transformer``, change all the fields of the ``sdtype``
+        to use the given transformer.
+
+        Args:
+            sdtype (str):
+                Semantic data type for the ``transformer``.
+            transformer (rdt.transformers.BaseTransformer):
+                Transformer to be used for the given ``sdtype``.
+        """
+        if not self.field_data_types:
+            msg = (
+                'Error: Nothing to update. Use the `detect_initial_config` method to'
+                'pre-populate all the sdtypes and transformers from your dataset.'
+            )
+            print(msg)  # noqa: T001
+
+        else:
+            for field, field_sdtype in self.field_data_types.items():
```

Comment:
I believe the desire is for all fields of the `sdtype` to now use the specified transformer. This means we should change everything in the `field_transformers` as well. On top of that, we should change the default for the `sdtype` because if for some reason a field hasn't been learned yet, the default will be used for it during fit.